lib/src/revset.rs

```diff
@@ -340,11 +343,15 @@ pub enum RevsetExpression {
     Symbol(String),
     Parents(Rc<RevsetExpression>),
     Children(Rc<RevsetExpression>),
-    Ancestors(Rc<RevsetExpression>),
+    Ancestors {
+        base: Rc<RevsetExpression>,
```

Comment:
nit: I'd prefer to call this `heads` unless there's a reason not to. I know they're not necessarily heads (you can do e.g. `:(A | A-)`).